BuyPolish/Pola/Manager/Analytics/AnalyticsRegistrant.swift

```diff
@@ -0,0 +1,16 @@
+import Firebase
+import Swinject
+
+final class AnalyticsRegistrant: DependencyRegistrant {
+    func registerDependency(container: Container) {
+        if FirebaseApp.isFirebaseAvailable {
+            container.register(AnalyticsProvider.self) { _ in
+                FirebaseAnalyticsProvider()
+            }
+        } else {
+            container.register(AnalyticsProvider.self) { _ in
+                ConsoleAnalyticsProvider()
+            }
+        }
```

Comment:
W tym momencie, zarówno `ConsoleAnalyticsProvider` jak i `FirebaseAnalyticsProvider` może zostać stworzone kilkukornie. Wydaje mi się, że lepiej by było to ograniczyć używając `.inObjectScope(.container)` który zapewni, że będzie 1 instancja danej klasy współdzielona w aplikacji - [więcej info](https://github.com/Swinject/Swinject/blob/master/Documentation/ObjectScopes.md). Być może inne implementacje `DependencyRegistrant` również tego potrzebują.

Reply:
Dzięki. Zastanawiałem się nad tym w trakcie, ale uznałem, że nie kilka obiektów nie zaszkodzi, a nie wiedziałem, że łatwo można to ogarnąć :)